src/modm/platform/uart/sam/usart/module.lb

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2021, Jeff McBride
+#
+# This file is part of the modm project.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+# -----------------------------------------------------------------------------
+
+def _get_properties(env, instance=None):
+    device = env[":target"]
+    peripheral = "Flexcom" if device.has_driver("flexcom") else "Usart"
+    return {
+        "type" : "usart",
+        "peripheral" : peripheral,
+        "id" : instance,
+        "prefix" : "US",
+        "sr" : "CSR",
+        "target" : device.identifier
+    }
+
+
+class Instance(Module):
+    def __init__(self, driver, instance):
+        self.driver = driver
+        self.instance = int(instance)
+
+    def init(self, module):
+        module.name = str(self.instance)
+        module.description = "Instance {}".format(self.instance)
+
+    def prepare(self, module, options):
+        module.depends(":platform:usart")
+
+        module.add_option(
+            NumericOption(
+                name="buffer.tx",
+                description="Size of transmit buffer",
+                minimum=1, maximum=2 ** 16 - 2,
+                default=64))
+        module.add_option(
+            NumericOption(
+                name="buffer.rx",
+                description="Size of receive buffer",
+                minimum=1, maximum=2 ** 16 - 2,
```

Comment:
```suggestion
                minimum=1, maximum="64K-2",
```
and above too. (Recently added to lbuild)

Is the minimum really 1? Ie. is unbuffered operation not implemented?

Reply:
Yes, it's not implemented in that driver. The other SAM driver, the sercom UART, only implements unbuffered mode. UART driver refactoring was on my list initially but this PR got huge enough. I'll do it after this is merged. Also the SAMx7x USB and external crystal support.